core/key.go

```diff
@@ -106,6 +107,13 @@ type CollectionSchemaVersionKey struct {
 
 var _ Key = (*CollectionSchemaVersionKey)(nil)
 
+type CollectionIndexKey struct {
+	CollectionID string
+	IndexName    string
```

Comment:
suggestion: I think we should use the Ids here, and those are uint32s not strings.

Reply:
kinda followed the lead of `CollectionKey`:
```go
type CollectionKey struct {
	CollectionName string
}
```

Reply:
I think that is the only place we persist collection name like this - it is an index to the collection CID using name to easily identify clashes.  All the other persisted keys use the collectionID I think.